### DISCUSSION TITLE
Flux ImageRepository secret type Opaque

### DISCUSSION THREAD

**Discussion by MiniMarPaz** [fluxcd/flux2#4822] [Q&A]
Hello.

I'm trying to set up Flux to automatically pull images from my image registry , but while setting up ImageRepository I'm getting error, that 
`failed to configure authentication options: unknown secret type "Opaque"` . In flux documentation I did not find that some other secret type should be used. Maybe someone stumbled upon similar issue ?  

Thank You!

<img width="1553" alt="image" src="https://github.com/fluxcd/flux2/assets/123905528/c6108b2b-81a6-43a3-88da-8dee724042ea">

**Comment by mloaf** [comment]
I met this problem too, have you found out the solution yet?

**Reply by mloaf** [reply]
Found the answer, instead of using a `generic` secret, we must use a `docker-registry` secret, my bad

**Comment by stefanprodan** [comment]
The Kubernetes Secret type must be `kubernetes.io/dockerconfigjson` and must contain the Docker config. Docs here: https://fluxcd.io/flux/components/image/imagerepositories/#secret-reference

**Reply by mloaf** [reply]
Thank you ~
